s3.py

```diff
@@ -540,12 +603,56 @@ def multipart_upload(self, representation, mirror_to, upload_class=MultipartS3Up
 MirrorUploader.IMPLEMENTATION_REGISTRY[S3Uploader.NAME] = S3Uploader
 
 
+class MinIOUploader(S3Uploader):
+    NAME = ExternalIntegration.MINIO
+
+    ENDPOINT_URL = 'ENDPOINT_URL'
+
+    SETTINGS = [
+        {
+            'key': ENDPOINT_URL,
+            'label': _('Endpoint URL'),
+            'description': _('MinIO\'s endpoint URL'),
+            'required': True
+        }
+    ] + S3Uploader.SETTINGS
+
+    def __init__(self, integration, client_class=None):
+        """Instantiate an S3Uploader from an ExternalIntegration.
+
+        :param integration: An ExternalIntegration
+
+        :param client_class: Mock object (or class) to use (or instantiate)
+            instead of boto3.client.
+        """
+        endpoint_url = integration.setting(
+            self.ENDPOINT_URL).value
+
+        _, host, _, _, _ = urlsplit(endpoint_url)
+
+        if not client_class:
+            client_class = boto3.client
+
+        if callable(client_class):
+            client_class = functools.partial(client_class, endpoint_url=endpoint_url)
+        else:
+            self.client = client_class
+
+        super(MinIOUploader, self).__init__(integration, client_class, host)
+
+
+# MirrorUploader.implementation will instantiate an S3Uploader
+# for storage integrations with protocol 'Amazon S3'.
```

Comment:
This comment was copy-and-pasted and should be updated.

Reply:
Updated the comment